log/std/logger.go

```diff
@@ -0,0 +1,224 @@
+// Package std is the implementation of the logger interface with the standard log package.
+package std
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	patronLog "github.com/beatlabs/patron/log"
+)
+
+var levelMap = map[patronLog.Level]string{
+	patronLog.DebugLevel: "DBG",
+	patronLog.InfoLevel:  "INF",
+	patronLog.WarnLevel:  "WRN",
+	patronLog.ErrorLevel: "ERR",
+	patronLog.FatalLevel: "FTL",
+	patronLog.PanicLevel: "PNC",
+}
+
+// Logger implementation of the std log.
+type Logger struct {
+	level      patronLog.Level
+	fields     map[string]interface{}
+	fieldsLine string
+	out        io.Writer
+	debug      *log.Logger
+	info       *log.Logger
+	warn       *log.Logger
+	error      *log.Logger
+	panic      *log.Logger
+	fatal      *log.Logger
+}
+
+// New constructor.
+func New(out io.Writer, lvl patronLog.Level, fields map[string]interface{}) *Logger {
+	fieldsLine := createFieldsLine(fields)
+
+	return &Logger{
+		out:        out,
+		debug:      createLogger(out, patronLog.DebugLevel, fieldsLine),
+		info:       createLogger(out, patronLog.InfoLevel, fieldsLine),
+		warn:       createLogger(out, patronLog.WarnLevel, fieldsLine),
+		error:      createLogger(out, patronLog.ErrorLevel, fieldsLine),
+		panic:      createLogger(out, patronLog.PanicLevel, fieldsLine),
+		fatal:      createLogger(out, patronLog.FatalLevel, fieldsLine),
+		level:      lvl,
+		fields:     fields,
+		fieldsLine: fieldsLine,
+	}
+}
+
+func createFieldsLine(fields map[string]interface{}) string {
+	if len(fields) == 0 {
+		return ""
+	}
+
+	// always return the fields in the same order
+	keys := make([]string, 0, len(fields))
+	for key := range fields {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+
+	sb := strings.Builder{}
+
+	for _, key := range keys {
```

Comment:
maybe i am missing something, but would it make sense to use only one fmt.Sprintf for everything ?

Reply:
the string builder is more performant. if you have to concatenate 2-3 things concatenation with + is even faster...